Move 4 functions from salt.utils to salt.utils.versions

terminalmage opened the pull request:
This moves the following functions from `salt.utils` to `salt.utils.versions`:

- `warn_until`
- `kwargs_warn_until`
- `compare_versions` (as `salt.utils.versions.compare`)
- `version_cmp`

terminalmage committed:
Move 4 functions from salt.utils to salt.utils.versions

This moves the following functions from salt.utils to salt.utils.versions:

- warn_until
- kwargs_warn_until
- compare_versions (as salt.utils.versions.compare)
- version_cmp

salt-jenkins commented:
@terminalmage, thanks for your PR! By analyzing the history of the files in this pull request, we identified @s0undt3ch, @JensRantil and @cachedout to be potential reviewers.

JensRantil commented:
I'll pass on this review, thanks.

rallytime commented:
@terminalmage Looks like the tests aren't very happy here I am afraid. Can you take a look?

terminalmage commented:
@rallytime Already in the middle of doing so ;)

terminalmage commented:
Go Go Jenkins!

terminalmage commented:
@rallytime there are still some issues here, but I'm going to pause on this PR and come back to it once I've sorted out the jenkins failures in 2016.11 and 2017.7.

terminalmage committed:
Guard raet import in try/except

This avoids a traceback upon import of salt.utils.raetevent in the
manage runner, when raet is not installed.

rallytime commented on salt/states/pkg.py:
```diff
@@ -80,10 +80,11 @@
+import salt.utils  # Can be removed once gen_state_tag is moved
 import salt.utils.pkg
 import salt.utils.platform
+import salt.utils.versions
```
Where is this import change used? I don't see any `salt.utils.versions.x` functions anywhere. I do see `from salt.utils.versions import LooseVersion` below the windows gated section, but that's all. Am I missing something?

terminalmage replied:
I missed making an update to one of the refs to `salt.utils.compare_versions`. Just pushed a commit to address this. Good catch!

rallytime replied:
👍

rallytime reviewed (commented):
@terminalmage I only had one small comment/question.

terminalmage committed:
Fix missed reference to salt.utils.compare_versions
